redfish-core/lib/processor.hpp

```diff
@@ -803,12 +805,13 @@ inline void getProcessorObject(const std::shared_ptr<bmcweb::AsyncResp>& resp,
         for (const auto& [objectPath, serviceMap] : subtree)
         {
             // Ignore any objects which don't end with our desired cpu name
-            if (!objectPath.ends_with(processorId))
+            sdbusplus::message::object_path path(objectPath);
+            std::string name = path.filename();
+            if (name.empty() || name != processorId)
```

Comment:
`procesorId` is not empty, and thus I think it is sufficient by doing 
```
if (name != processorId)
```

Reply:
Agreed, I went ahead and made this change.